cache-prefetch.go

```diff
@@ -0,0 +1,183 @@
+package rdns
+
+import (
+	"errors"
+	"expvar"
+	"github.com/miekg/dns"
+	"github.com/sirupsen/logrus"
+	"sync"
+	"time"
+)
+
+type CachePrefetch struct {
+	CachePrefetchOptions
+	id       string
+	resolver Resolver
+	mu       sync.Mutex
+	metrics  *CachePrefetchMetrics
+}
+
+
+type CachePrefetchMetrics struct {
+	// Cache hit count.
+	domainEntries map[string]CachePrefetchEntry
+}
+type CachePrefetchEntry struct {
+	// request hit count.
+	hit 			  expvar.Int
+
+	// 0 for no prefetching job
+	// 1 for prefetching job is active
+	// 2 for stopped by error
+	prefetchingStatus expvar.Int
```

Comment:
Instead of documenting what each value means, you can use a dedicated type.

```go
type PrefetchState int

const (
  PrefetchStateNone = iota
  PrefetchStateActive
  PrefetchStateOther
)
```

Reply:
I like this better